sharktank/tests/kernels/wave/extend_attention_test.py

```diff
@@ -261,9 +295,167 @@ def test_no_cache(
 
         v_noise = v * 0.05
         extend_attention_v_noise = ops.extend_attention(
-            q=q, k=k, v=v_noise, start_positions=start_positions, seq_lens=seq_lens
+            q=q,
+            k=k,
+            v=v_noise,
+            kv_cache=None,
+            k_indices=indices_no_cache,
+            v_indices=indices_no_cache,
+            start_positions=start_positions,
+            seq_lens=seq_lens,
         )
         with pytest.raises(AssertionError):
             torch.testing.assert_close(
                 sdpa, extend_attention_v_noise, atol=1e-3, rtol=1e-3
             )
+
+    @pytest.mark.skipif(not torch.cuda.is_available(), reason="Needs CUDA/HIP device.")
+    @pytest.mark.parametrize(
+        "batch, heads, seq_len, head_dim, attn_dtype, device",
+        [
+            (1, 4, 32, 16, torch.float16, "cuda"),
+        ],
+    )
+    def test_extend_kv_cache_single_request_two_chunks(
+        self,
+        batch,
+        heads,
+        seq_len,
+        head_dim,
+        attn_dtype,
+        device,
+    ):
```

Comment:
When you are writing this much boiler plate for a test its a sign that either helpers / cleanup should be included. The transposition, data setup, etc inherently means its too complex for clearly describe the test.